src/main/java/net/javadiscord/javabot/systems/moderation/ModerationService.java

```diff
@@ -198,21 +201,43 @@ public boolean removeTimeout(Member member, String reason, Member removedBy, Mes
 	 * @param bannedBy The member who is responsible for banning this member.
 	 * @param channel  The channel in which the ban was issued.
 	 * @param quiet    If true, don't send a message in the channel.
-	 * @return Whether the moderator has the permission to ban this member or not.
 	 */
-	public boolean ban(Member member, String reason, Member bannedBy, MessageChannel channel, boolean quiet) {
+	public void ban(Member member, String reason, Member bannedBy, MessageChannel channel, boolean quiet) throws IllegalArgumentException, UserPermissionException {
 		var banEmbed = buildBanEmbed(member, reason, bannedBy);
 		if (canBanUser(member, bannedBy)) {
 			member.getUser().openPrivateChannel().queue(
 					c -> c.sendMessage(config.getBanMessageText()).setEmbeds(banEmbed).queue(),
 					e -> log.info("Could not send Ban Direct Message to User {}", member.getUser().getAsTag())
 			);
-			bannedBy.getGuild().ban(member, BAN_DELETE_DAYS, reason).queue();
+
+			try {
+				bannedBy.getGuild().ban(member, BAN_DELETE_DAYS, reason).queue();
+			} catch (InsufficientPermissionException | HierarchyException e) {
+				throw new UserPermissionException(channel.getJDA().getSelfUser(), e);
+			}
+
 			config.getLogChannel().sendMessageEmbeds(banEmbed).queue();
 			if (!quiet) channel.sendMessageEmbeds(banEmbed).queue();
-			return true;
+		} else {
+			throw new UserPermissionException(bannedBy.getUser(), Permission.BAN_MEMBERS, "User does not have permissions to ban the targeted member.");
 		}
-		return false;
+	}
+
+	/**
+	 * Bans a member.
+	 *
+	 * @param snowFlakeID The member to ban.
+	 * @param reason      The reason for banning the member.
+	 * @param bannedBy    The member who is responsible for banning this member.
+	 * @param channel     The channel in which the ban was issued.
+	 * @param quiet       If true, don't send a message in the channel.
+	 */
+	public void ban(String snowFlakeID, String reason, Member bannedBy, MessageChannel channel, boolean quiet) throws IllegalArgumentException, InsufficientPermissionException, HierarchyException {
```

Comment:
maybe rename `snowFlakeID` to `targetId` and make it a long instead of a String

Reply:
No, as explained in the comment below. 

https://github.com/Java-Discord/JavaBot/pull/239#discussion_r825337024